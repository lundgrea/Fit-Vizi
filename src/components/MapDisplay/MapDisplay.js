import React, { Component } from 'react';
import { Map, GoogleApiWrapper, Polyline, Marker } from 'google-maps-react';

const API_KEY =`${process.env.REACT_APP_API_KEY}`



export class MapDisplay extends Component { 


  displayPolyline = () => {
    let routes = this.generatePolylines()
    return (
      <Polyline
        path={routes}
        strokeColor="#DB3A34"
        strokeOpacity={0.8}
        strokeWeight={5} 
      />
    )
  }

  generateMarker = (object) => {
    console.log(object)
    if (this.props.selectedItem !== '') {
      return (
        <Marker 
          position={{
            lat: object.positionLat,
            lng: object.positionLong
          }}        
          title="Selected Data"
          id={1}
          draggable={true}
        />
      )
    }
  }

  generatePolylines = () => {
    let samples = this.props.workoutData
    let paths = samples.reduce((endValue, sample) => {
      if (sample.positionLat !== undefined || sample.positionLong !== undefined) {
        let object = {lat: '', lng: ''}
        object.lat = sample.positionLat
        object.lng = sample.positionLong
        endValue.push(object)
      return endValue
      } else {
      return endValue
      }
    }, [{lat: 40.01488, lng: -105.131}])
    return paths
  }


  render() {
    return (
      <div className='map'>
        <h2>Map of Travel Route</h2>
        <Map
          google={this.props.google}
          zoom={12}
          style={mapStyles}
          initialCenter={{ lat: 40.01488, lng: -105.131}}
        >
        {this.generateMarker(this.props.selectedItem)}
        {this.displayPolyline()}
        </Map>
      </div>
    );
  } 
}

export default GoogleApiWrapper({
  apiKey: API_KEY
})(MapDisplay);


const mapStyles = {
  width: '75%',
  height: '75%',
};