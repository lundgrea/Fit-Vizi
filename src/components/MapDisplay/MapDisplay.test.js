

import React from 'react';
import { shallow } from 'enzyme';
import { MapDisplay } from './MapDisplay';

describe('MapDisplay', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<MapDisplay 
      workoutData={[]}
      selectedItem={'positionLat', 'positionLong'}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});