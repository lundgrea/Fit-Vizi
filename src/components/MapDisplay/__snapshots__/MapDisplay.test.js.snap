// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDisplay should match the snapshot with all data passed in correctly 1`] = `
<div
  className="map"
>
  <h2>
    Map of Travel Route
  </h2>
  <Map
    center={Object {}}
    centerAroundCurrentLocation={false}
    containerStyle={Object {}}
    initialCenter={
      Object {
        "lat": 40.01488,
        "lng": -105.131,
      }
    }
    style={
      Object {
        "height": "75%",
        "width": "75%",
      }
    }
    visible={true}
    zoom={12}
  >
    <Polyline
      name="Polyline"
      path={
        Array [
          Object {
            "lat": 40.01488,
            "lng": -105.131,
          },
        ]
      }
      strokeColor="#DB3A34"
      strokeOpacity={0.8}
      strokeWeight={5}
    />
  </Map>
</div>
`;
